vars/pipeline2ATX.groovy

```diff
@@ -351,18 +341,16 @@ def calculateErrorTime(executionTestSteps) {
  * Determines the respective phase of the stage being examined.
  *
  * @param stageName
- *      the name of the current stage
- * @param currentPhase
- *      the corresponding phase in which the current stage is located
+ *      the name of the stage
  * @return the current phase
  */
-def getCurrentPhase(stageName, currentPhase) {
-    if (stageName.contains("stage (Declarative: Post")) {
+def getStagePhase(stageName) {
+    if (stageName.contains("Post Actions") || stageName.contains("Post:")) {
         return 'teardown'
-    } else if (!stageName.contains("stage (Declarative") && currentPhase == 'setup') {
-        return 'execution'
+    } else if (stageName.contains("Declarative:") || stageName.contains("Pre:")) {
+        return 'setup'
     }
-    return currentPhase
+    return 'execution'
 }
```

Comment:
This is fine as long as we are sure that no other step name is ever present in the step arguments

our stage name is "$name ($arguments)" where name seems to be 'stage'
is the first argument always the name or not ? if so we could use startsWith instead of contains

Reply:
This is defined by us during the read/conversion of the flowtable into test steps. The format is `stage ($stagename)`.